docs/node-mixin/dashboards/use.libsonnet

```diff
@@ -1,275 +1,467 @@
-local g = import 'github.com/grafana/jsonnet-libs/grafana-builder/grafana.libsonnet';
+local grafana = import 'github.com/grafana/grafonnet-lib/grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+local row = grafana.row;
+local prometheus = grafana.prometheus;
+local template = grafana.template;
+local graphPanel = grafana.graphPanel;
+
+local c = import '../config.libsonnet';
+
+local datasourceTemplate = {
+  current: {
+    text: 'Prometheus',
+    value: 'Prometheus',
+  },
+  hide: 0,
+  label: null,
+  name: 'datasource',
+  options: [],
+  query: 'prometheus',
+  refresh: 1,
+  regex: '',
+  type: 'datasource',
+};
+
+local clusterTemplate =
+  template.new(
+    name='cluster',
+    datasource='$datasource',
+    query='label_values(node_cpu_seconds_total, %s)' % c._config.clusterLabel,
```

Comment:
node_cpu_seconds_total is a particular expensive one. node_time_seconds would be better

Reply:
For some reason this uses 'cluster' even though I've set clusterLabel to `kube_cluster` which works fine in the actual graph query expressions:
```
$ grep label_val.*node_cpu_seconds_total build/node_exporter/ -R
build/node_exporter//dashboards/node-rsrc-use.json:        "query": "label_values(node_cpu_seconds_total, cluster)",
build/node_exporter//dashboards/node-cluster-rsrc-use.json:        "query": "label_values(node_cpu_seconds_total, cluster)",
$ grep kube_cluster build/node_exporter/ -R|head -1
build/node_exporter//dashboards/node-rsrc-use.json:              "expr": "instance:node_cpu_utilisation:rate180s{job=\"node\", instance=\"$instance\", kube_cluster=\"$cluster\"} != 0",
```

Reply:
That is strange 🤔. I just tried myself and got this result:

![image](https://user-images.githubusercontent.com/24193764/126215414-b173d806-b2b5-48a8-aa9c-a921fe16c558.png)

Is it possible that you forgot to update the dependencies?

Reply:
Ah, okay yeah changing config.libsonnet works but I want to import it in a different jsonnet file and override the config which doesn't work:
```
$ echo '(import "./mixin.libsonnet" + { _config+:: { clusterLabel: "foo" } }).grafanaDashboards' | jsonnet -J vendor/ -J . /dev/stdin|grep -E 'node_time_seconds|foo='|tail -2
               "query": "label_values(node_time_seconds, cluster)",
               "query": "label_values(node_exporter_build_info{job=\"node\", foo=\"$cluster\"}, instance)",
```

To make that possible, instead of reading the config directly here: https://github.com/prometheus/node_exporter/pull/2017/files#diff-23d2fdec9ecd007ba3109e98b153df62310bcb3b2ea03c01c0d3dd0c1ecee296R29 we'd need to use $ to access the _config at the root element

Reply:
I think you'd need to make clusterTemplate a hidden field in the document instead of a local variable.

Reply:
@ArthurSens this should do it:
```
diff --git a/docs/node-mixin/dashboards/use.libsonnet b/docs/node-mixin/dashboards/use.libsonnet
index 3a8959d..0121947 100644
--- a/docs/node-mixin/dashboards/use.libsonnet
+++ b/docs/node-mixin/dashboards/use.libsonnet
@@ -22,18 +22,6 @@ local datasourceTemplate = {
   type: 'datasource',
 };
 
-local clusterTemplate =
-  template.new(
-    name='cluster',
-    datasource='$datasource',
-    query='label_values(node_time_seconds, %s)' % c._config.clusterLabel,
-    current='',
-    hide=if c._config.showMultiCluster then '' else '2',
-    refresh=2,
-    includeAll=false,
-    sort=1
-  );
-
 local CPUUtilisation =
   graphPanel.new(
     'CPU Utilisation',
@@ -142,6 +130,17 @@ local diskSpaceUtilisation =
   ) { tooltip+: { sort: 2 } };
 
 {
+  _clusterTemplate:: template.new(
+    name='cluster',
+    datasource='$datasource',
+    query='label_values(node_time_seconds, %s)' % $._config.clusterLabel,
+    current='',
+    hide=if $._config.showMultiCluster then '' else '2',
+    refresh=2,
+    includeAll=false,
+    sort=1
+  ),
+
   grafanaDashboards+:: {
                          'node-rsrc-use.json':
 
@@ -154,7 +153,7 @@ local diskSpaceUtilisation =
                              graphTooltip='shared_crosshair'
                            )
                            .addTemplate(datasourceTemplate)
-                           .addTemplate(clusterTemplate)
+                           .addTemplate($._clusterTemplate)
                            .addTemplate(
                              template.new(
                                'instance',
@@ -219,7 +218,7 @@ local diskSpaceUtilisation =
                              graphTooltip='shared_crosshair'
                            )
                            .addTemplate(datasourceTemplate)
-                           .addTemplate(clusterTemplate)
+                           .addTemplate($._clusterTemplate)
                            .addRow(
                              row.new('CPU')
                              .addPanel(

```

Reply:
Thanks for noticing that, I haven't tested by consuming the mixing in another jsonnet file until now.

It think it should be good now, but could you take another look?

Reply:
You missed the $._config.showMultiCluster, see my last commit.
Unfortunately I forgot to sign-off and can't amend since it's in your fork (tbh surprised I could edit the file here in the web ui in the first place). Anyway, created https://github.com/prometheus/node_exporter/pull/2101 to include your changes and that fix.